scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,81 @@ def visit_compare(self, node):
                 self.add_message('inequality-with-none', node=node)
 
 
+class DisallowedFunctionsChecker(checkers.BaseChecker):
+    """Custom pylint checker for language specific general purpose
+    regex checks of functions calls to be removed or replaced.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallowed-function-calls'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Please do not use %s.',
+            'remove-disallowed-function-calls',
+            'Please remove disallowed function calls.',
+        ),
+        'C0032': (
+            'Please replace the call to %s with %s.',
+            'replace-disallowed-function-calls',
+            (
+                'Please replace disallowed function'
+                'calls with allowed alternative.'),
+        ),
+    }
+    python_utils_file = b'python_utils.py'
+    python_utils_test_file = b'python_utils_test.py'
+
+    options = (
+        (
+            'disallowed-functions',
+            {
+                'default': (),
+                'type': 'csv',
+                'metavar': '<comma separated list>',
+                'help': (
+                    'List of strings of disallowed function names. '
+                    'Strings should be either in the format \'A=>B\', '
+                    'where A is the disallowed function and B is the '
+                    'replacement or in the format \'A\', which signifies '
+                    'that A should just be removed.')
+            }
+        ),)
+
+    def visit_call(self, node):
+        """Visit a function call to ensure that the call is
+        not using any disallowed functions.
+
+        Args:
+            node: astroid.nodes.Call. Node to access call content.
+        """
+        disallowed_entries = self.config.disallowed_functions
+
+        func = node.func
+        for entry in disallowed_entries:
+            splits = [s.strip() for s in entry.split('=>')]
+            assert len(splits) > 0 and len(splits) <= 2
+            func_name = splits[0]
+            if (
+                    (hasattr(func, 'attrname') and func.attrname == func_name)
+                    or (hasattr(func, 'name') and func.name == func_name)):
+                if len(splits) == 1:
+                    self.add_message(
+                        'remove-disallowed-function-calls',
+                        node=node, args=func_name)
+                else:
+                    replacement_msg = splits[1]
+                    if (
+                            'python_utils' in node.root().file
+                            and 'python_utils' in replacement_msg):
```

Comment:
Why do we need this check? (Note this file should be independent of anything in the oppia codebase.)

Reply:
Removed!